src/verify_cert.rs

```diff
@@ -19,12 +19,109 @@ use crate::{
+    ///
+    /// By default revocation checking will fail if the revocation status of a certificate cannot
+    /// be determined. This can be customized using the [allow_unknown_status] method.
+    pub fn new(crls: &'a [&'a dyn CertRevocationList]) -> Result<Self, CrlsRequired> {
+        match crls.len() {
+            0 => Err(CrlsRequired),
+            _ => Ok(Self {
+                crls,
+                depth: RevocationCheckDepth::Chain,
+                status_requirement: RevocationStatusRequirement::ForbidUnknown,
+            }),
+        }
+    }
+
+    /// Customize the depth at which revocation checking will be performed, controlling
+    /// whether only the end-entity (leaf) certificate in the chain to a trust anchor will
+    /// have its revocation status checked, or whether the intermediate certificates will as well.
+    pub fn with_depth(mut self, depth: RevocationCheckDepth) -> Self {
+        self.depth = depth;
+        self
+    }
+
+    /// Treat unknown revocation status permissively, acting as if the certificate were not
+    /// revoked.
+    pub fn allow_unknown_status(mut self) -> Self {
+        self.status_requirement = RevocationStatusRequirement::AllowUnknown;
+        self
+    }
+
+    /// Treat unknown revocation status strictly, considering it an error condition.
+    pub fn forbid_unknown_status(mut self) -> Self {
+        self.status_requirement = RevocationStatusRequirement::ForbidUnknown;
+        self
+    }
+
+    /// Construct a [RevocationOptions] instance based on the builder's configuration.
+    pub fn build(self) -> RevocationOptions<'a> {
+        RevocationOptions {
+            crls: self.crls,
+            depth: self.depth,
+            status_requirement: self.status_requirement,
+        }
+    }
+}
+
+/// Describes how revocation checking is performed, if at all. Can be constructed with a
+/// [RevocationOptionsBuilder] instance.
+#[derive(Debug, Copy, Clone)]
+pub struct RevocationOptions<'a> {
+    crls: &'a [&'a dyn CertRevocationList],
+    depth: RevocationCheckDepth,
+    status_requirement: RevocationStatusRequirement,
+}
+
+/// Describes how much of a certificate chain is checked for revocation status.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum RevocationCheckDepth {
+    /// Only check the end entity (leaf) certificate's revocation status.
+    EndEntity,
+    /// Check the revocation status of the end entity (leaf) and all intermediates.
+    Chain,
+}
+
+/// Describes how to handle the case where a certificate's revocation status is unknown.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum RevocationStatusRequirement {
+    /// Treat unknown revocation status permissively, acting as if the certificate were
+    /// not revoked.
+    AllowUnknown,
+    /// Treat unknown revocation status as an error condition, yielding
+    /// [Error::UnknownRevocationStatus].
+    ForbidUnknown,
```

Comment:
Should we use `Deny` here instead of `Forbid`? Maybe hoist the `Unknown` into the type name instead of repeating it in the variants? (Suggestion: `UnknownStatusPolicy`?)

Reply:
That sounds good. I think having shorter variant names makes sense. Updated.